### ISSUE TITLE
Multiline text bubble override produces huge margins around text

### ISSUE THREAD

**Issue by Cjreek** [Beamdog/nwn-issues#681]
### Describe the bug

Adding new lines (\n) produces a huge margin around text bubbles. Natural line breaks don't cause this issue:

![image](https://github.com/user-attachments/assets/7284fd9d-4ad6-4c5e-8c0f-f62b790813c7)

without manual line breaks the margin stays constant:

![image](https://github.com/user-attachments/assets/4768d406-61b9-483a-a5ad-859c18108e0d)

Code:
```nwscript
void main()
{
    object oItem = GetObjectByTag("TEST_SWORD"); 

    string sItemTooltip = "Frostwind\n";
    sItemTooltip += "Cryptic Sword\n";
    sItemTooltip += "One-Hand Damage: 16 to 251\n";
    sItemTooltip += "Durability: 44 of 44\n";
    sItemTooltip += "Required Dexterity: 109\n";
    sItemTooltip += "Required Strength: 99\n";
    sItemTooltip += "Required Level: 70\n";
    sItemTooltip += "Sword Class - Very Fast Attack Speed\n";
    sItemTooltip += "+25% increased Attack Speed\n";
    sItemTooltip += "+227% Enhanced Damage\n";
    sItemTooltip += "Adds 237-486 Cold Damage\n";
    sItemTooltip += "+10 to Arctic Blast\n";
    sItemTooltip += "Freezes Target +4\n";
    sItemTooltip += "Cold Absorb 15%\n";
    sItemTooltip += "Half Freeze Duration";

    SetObjectUiDiscoveryMask(oItem, OBJECT_UI_DISCOVERY_TEXTBUBBLE_MOUSEOVER);
    SetObjectTextBubbleOverride(oItem, OBJECT_UI_TEXT_BUBBLE_OVERRIDE_REPLACE, sItemTooltip);
}
```

### Game version

37-11

### OS

Linux

### Distribution

Steam

**Comment by tinygiant98** [comment]
Non-NUI tooltips use various bubble "models" for the shaded background.  There are 15 of these models of various heights, but there are only three widths: 100, 150 and 300.  This means that any lines that are less than 150 will use a model that is 150 or less, but any lines that area greater than 150, even by a little, will use a model that is 300 wide. 

In this picture, you'll see a tooltip that is just under the 150 limit:

<img width="421" height="306" alt="Image" src="https://github.com/user-attachments/assets/46e5cba4-2356-4d02-a748-e98dbe2390c4" />

This is the same tooltip with one additional character, putting it over the 150 limit:

<img width="686" height="314" alt="Image" src="https://github.com/user-attachments/assets/837d10d4-d5ce-4130-8042-1640490c4aa6" />

This same concept applies to the height, however, there are many more heights available.  Unfortunately, as you get to taller tooltip models, the heights grow at greater rates.  Here's an example of the height progression for the 300 models:  20 -> 34 -> 48 -> 62 -> 76 -> 104 -> 146 -> 230 -> 454.  So as the number of lines grows, the height of the textbox grows quickly.

These two factors give the *appearance* that there are ridiculously large margins, but this is a limitation of the method involved in displaying tooltips/bubbles.

One method to fix this, using the current implementation, is to add additional models to the game.  @Finaldeath may have ideas on additional methods to fix.

ref:  `guipaneltextbubble.cpp:52-81`
models: `aurora_gui.bif:pnl_bubble#_###.mdl`

**Comment by Cjreek** [comment]
I'm amazed. I don't really know what to say 😆 
I guess I'd still appreciate _something_ done here, but even knowing what it actually does and the numbers might help to work around it a little bit

...

**Comment by Finaldeath** [comment]
It basically is a lookup like tinygiant said. We could easily put in a number of additional MDL Files and add to the lookup table, it should just use them then. Add some more for the intermediate values perhaps.
